### PR TITLE
[LM-10923] Added API Key Security

### PR DESCRIPTION
**Pull request by ishaan1607**
## Description

**Features**
- Added API Key Security

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: app/src/main/java/com/likeminds/chatsampleapp/GetTokensTask.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    private lateinit var authPreferences: AuthPreferences
+
+    // Get tokens from the server
+    suspend fun getTokens(context: Context, isProd: Boolean): Pair<String, String> {
+        return withContext(Dispatchers.IO) {
+            //get api url
+            val apiUrl = if (isProd) {
+                "https://auth.likeminds.community/sdk/initiate"
+            } else {
+                "https://betaauth.likeminds.community/sdk/initiate"
```
Comment (sidcr7-likeminds) [inline app/src/main/java/com/likeminds/chatsampleapp/GetTokensTask.kt:25]: store these values in `BuildConfig`

File: chatmm/src/main/java/com/likeminds/chatmm/member/util/UserResponseConvertor.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    //convert validate user response to user response
+    fun getUserResponse(validateUserResponse: ValidateUserResponse): UserResponse {
+        return UserResponse(
+            user = validateUserResponse.user,
+            community = validateUserResponse.community,
+            appAccess = validateUserResponse.appAccess
+        )
+    }
+}
```
Comment (sidcr7-likeminds) [inline chatmm/src/main/java/com/likeminds/chatmm/member/util/UserResponseConvertor.kt:29]: Can write this in `ViewDataonverter.kt`

File: chatmm/src/main/java/com/likeminds/chatmm/utils/user/LMChatUserMetaData.kt [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+        fun getInstance(): LMChatUserMetaData {
+            if (instance == null) {
+                instance = LMChatUserMetaData()
+            }
+            return instance!!
+        }
+    }
+
+    fun init(
```
Comment (sidcr7-likeminds) [inline chatmm/src/main/java/com/likeminds/chatmm/utils/user/LMChatUserMetaData.kt:36]: add comments

File: chatmm/src/main/java/com/likeminds/chatmm/utils/user/LMChatUserMetaData.kt [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        deviceId: String?
+    ) {
+        this.domain = domain
+        this.enablePushNotifications = enablePushNotifications
+        this.deviceId = deviceId
+
+        mChatClient = LMChatClient.getInstance()
+    }
+
+    fun onPostUserSessionInit(
```
Comment (sidcr7-likeminds) [inline chatmm/src/main/java/com/likeminds/chatmm/utils/user/LMChatUserMetaData.kt:48]: add comments

File: chatmm/src/main/java/com/likeminds/chatmm/utils/user/LMChatUserMetaData.kt [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            } catch (e: Exception) {
+                Log.w(
+                    SDKApplication.LOG_TAG,
+                    "Please add firebase to your project to enable notifications"
+                )
+            }
+        }
+    }
+
+    private fun registerDevice(token: String) {
```
Comment (sidcr7-likeminds) [inline chatmm/src/main/java/com/likeminds/chatmm/utils/user/LMChatUserMetaData.kt:144]: add comments

**Review by sidcr7-likeminds — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/likeminds/chatsampleapp/GetTokensTask.kt (hunk 1 above)
Comment (ishaan1607) [reply app/src/main/java/com/likeminds/chatsampleapp/GetTokensTask.kt:25]: These are example code, not necessaey

**Review by sidcr7-likeminds — APPROVED** [review]

**Pull request merged**